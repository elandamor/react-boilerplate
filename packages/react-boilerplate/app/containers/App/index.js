/**
 * App
 */

import React from 'react';
import { hot } from 'react-hot-loader';
// import PropTypes from 'prop-types';
import { Route, Switch, withRouter } from 'react-router-dom';
// Pages
import Home from '../../pages/Home/Loadable';
// Styled-Components
import Wrapper from './styles';

// eslint-disable-next-line react/prefer-stateless-function
class App extends React.Component {
  render() {
    return (
      <Wrapper>
        <Switch>
          <Route
            exact
            path="/"
            render={(props) => (
              <Home {...props} />
            )}
          />
        </Switch>
      </Wrapper>
    );
  }
}

App.propTypes = {

};

export default withRouter(hot(module)(App));
