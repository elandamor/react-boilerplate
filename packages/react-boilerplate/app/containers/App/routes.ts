// tslint:disable:object-literal-sort-keys
import Home from '../../pages/Home/Loadable';

const routes = [{
  path: '/',
  component: Home,
}];

export default routes;
