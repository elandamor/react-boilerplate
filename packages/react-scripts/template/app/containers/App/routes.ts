// tslint:disable:object-literal-sort-keys
import { Home } from '../../pages';

import { IRouteProps } from '../../components/Routes';

const routes: IRouteProps[] = [
  {
    exact: true,
    path: '/',
    component: Home,
  },
];

export default routes;
