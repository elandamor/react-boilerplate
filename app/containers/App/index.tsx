import React, { Component } from 'react';
import { Route, Switch, withRouter } from 'react-router-dom';
// Components
import Routes from '../../components/Routes';
// Routes
import routes from './routes';
// Styles
import Wrapper from './styles';

/**
 * @render react
 * @name App container
 * @description The skeleton around the actual pages, and should only
 * contain code that should be seen on all pages. (e.g. navigation bar).
 */

// eslint-disable-next-line react/prefer-stateless-function
class App extends Component<{}> {
  public render() {
    return (
      <Wrapper>
        <Switch>
          <Routes routes={routes} />
        </Switch>
      </Wrapper>
    );
  }
}

export default withRouter(App);
