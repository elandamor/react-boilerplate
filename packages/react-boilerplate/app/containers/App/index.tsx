import classNames from 'classnames';
import React, { Component } from 'react';
import Measure from 'react-measure';
import { RouteComponentProps, withRouter } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';
// Components
import { ErrorBoundary, Routes } from '../../components';
// Routes
import routes from './routes';
// Styles
import Wrapper from './styles';

export const breakpoints = (width: number) => {
  if (width < 600) {
    return 'v-xsmall';
  }
  if (width >= 600 && width < 1024) {
    return 'v-small';
  }
  if (width >= 1024 && width < 1440) {
    return 'v-medium';
  }
  if (width >= 1440 && width < 1920) {
    return 'v-large';
  }
  if (width >= 1920) {
    return 'v-xlarge';
  }
  return 'v-unknown';
};

/* tslint:disable:object-literal-sort-keys */
const themeLight = {
  isDark: false,
  palette: {
    bodyBackground: '#FAFAFA',
    brandPrimary: '#FFFFFF',
    cardBackground: '#FFFFFF',
    cardBorderColor: '#E4E6E9',
  },
};
/* tslint:enable:object-literal-sort-keys */

// import { makeDebugger } from '../../lib';
// const debug = makeDebugger('App');

export interface IProps extends RouteComponentProps<any> {}

interface IState {
  bounds: {
    height: number;
    width: number;
  };
  theme: object;
}

/**
 * @render react
 * @name App container
 * @description The skeleton around the actual pages, and should only
 * contain code that should be seen on all pages. (e.g. navigation bar).
 */
class App extends Component<IProps, IState> {
  protected componentIsMounted: boolean;
  protected previousLocation: object;

  constructor(props: IProps) {
    super(props);

    this.state = {
      bounds: {
        height: 0,
        width: 0,
      },
      theme: themeLight,
    };

    this.previousLocation = props.location;
  }

  public componentDidMount() {
    this.componentIsMounted = true;
  }

  public componentWillUpdate(nextProps: IProps) {
    const { location } = this.props;

    if (
      nextProps.history.action !== 'POP' &&
      (!location.state || !location.state.modal)
    ) {
      this.previousLocation = this.props.location;
    }
  }

  public componentWillUnmount() {
    this.componentIsMounted = false;
  }

  public setState(nextState: any, cb?: () => void) {
    if (this.componentIsMounted) {
      super.setState(nextState, cb);
    }
  }

  public render() {
    const { location } = this.props;
    const {
      bounds: { width },
    } = this.state;

    const isModal = !!(
      location.state &&
      location.state.modal &&
      this.previousLocation !== location
    );

    return (
      <ThemeProvider theme={this.state.theme}>
        <Measure
          bounds
          onResize={(contentRect) => {
            this.setState({ bounds: contentRect.bounds });
          }}
        >
          {({ measureRef }) => (
            <Wrapper
              className={classNames('c-app__container', breakpoints(width))}
              innerRef={measureRef}
            >
              <ErrorBoundary>
                <Routes
                  location={isModal ? this.previousLocation : location}
                  routes={routes}
                />
              </ErrorBoundary>
            </Wrapper>
          )}
        </Measure>
      </ThemeProvider>
    );
  }
}

export default withRouter(App);
