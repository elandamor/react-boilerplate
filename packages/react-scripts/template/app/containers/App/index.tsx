import classNames from 'classnames';
import React, { Component, Suspense } from 'react';
import Measure from 'react-measure';
import { RouteComponentProps, withRouter } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';
// Components
import { ErrorBoundary, LoadingBar, Routes } from '../../components';
// Contexts
import NetworkStatusProvider from '../../contexts/networkStatus.context';
// Routes
import routes from '../../routes';
// Styles
import GlobalStyles from '../../global-styles';
import Wrapper from './styles';

import { breakpoints, makeDebugger } from '../../utils';
const debug = makeDebugger('App');

const defaultTheme = {
  breakpoints: {
    small: '600px',
    medium: '1024px',
    large: '1440px',
    xLarge: '1920px',
  },
};

/* tslint:disable:object-literal-sort-keys */
const themeLight = {
  ...defaultTheme,
  isDark: false,
  palette: {
    bodyBackground: '#FAFAFA',
    brandPrimary: '#FFFFFF',
    cardBackground: '#FFFFFF',
    cardBorderColor: '#E4E6E9',
  },
};
/* tslint:enable:object-literal-sort-keys */

export interface IProps extends RouteComponentProps<any> {}

interface IState {
  bounds: {
    height: number;
    width: number;
  };
  theme: object;
}

/**
 * @render react
 * @name App container
 * @description The skeleton around the actual pages, and should only
 * contain code that should be seen on all pages. (e.g. navigation bar).
 */
class App extends Component<IProps, IState> {
  protected componentIsMounted: boolean;
  protected previousLocation: object;

  constructor(props: IProps) {
    super(props);

    this.state = {
      bounds: {
        height: 0,
        width: 0,
      },
      theme: themeLight,
    };

    this.previousLocation = props.location;
  }

  public componentDidMount() {
    this.componentIsMounted = true;
    debug('componentIsMounted');
  }

  public componentWillUpdate(nextProps: IProps) {
    const { location } = this.props;

    if (
      nextProps.history.action !== 'POP' &&
      (!location.state || !location.state.modal)
    ) {
      this.previousLocation = this.props.location;
    }
  }

  public componentWillUnmount() {
    this.componentIsMounted = false;
  }

  public setState(nextState: any, cb?: () => void) {
    if (this.componentIsMounted) {
      super.setState(nextState, cb);
    }
  }

  public render() {
    const { location } = this.props;
    const {
      bounds: { width },
    } = this.state;

    const isModal = !!(
      location.state &&
      location.state.modal &&
      this.previousLocation !== location
    );

    return (
      <ThemeProvider theme={this.state.theme}>
        <NetworkStatusProvider>
          <Measure
            bounds
            onResize={(contentRect) => {
              this.setState({ bounds: contentRect.bounds });
            }}
          >
            {({ measureRef }) => (
              <Wrapper
                className={classNames('c-app__container', breakpoints(width))}
                // @ts-ignore
                ref={measureRef}
              >
                <GlobalStyles />
                <ErrorBoundary>
                  <Suspense fallback={<LoadingBar loading />}>
                    <Routes
                      location={isModal ? this.previousLocation : location}
                      routes={routes}
                    />
                  </Suspense>
                </ErrorBoundary>
              </Wrapper>
            )}
          </Measure>
        </NetworkStatusProvider>
      </ThemeProvider>
    );
  }
}

export default withRouter(App);
