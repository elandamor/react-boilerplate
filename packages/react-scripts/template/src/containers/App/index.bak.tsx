import React, { Suspense } from 'react';
import { RouteComponentProps, withRouter } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';
import { Normalize } from 'styled-normalize';
import get from 'lodash/get';
import merge from 'lodash/merge';
import { useTransition } from 'react-spring';
// Components
import { ErrorBoundary, Header, LoadingBar, Routes, AnimatedWrapper } from '@app/components';
// Contexts
import NetworkStatusProvider from '@app/contexts/networkStatus.context';
// Routes
import routes from '@app/routes';
// Styles
import { Wrapper } from './styles';

import GlobalStyles from '@app/global-styles';

import baseTheme from '@app/theme';
import { useDarkMode, useRouter } from '@app/hooks';

// import { makeDebugger } from '@app/utils';
// const debug = makeDebugger('App');

export interface IAppProps extends RouteComponentProps {}

/**
 * @render react
 * @name App container
 * @description The skeleton around the actual pages, and should only
 * contain code that should be seen on all pages. (e.g. navigation bar).
 */

const App = () => {
  const { location } = useRouter();
  const [darkMode] = useDarkMode();

  // Merge the color mode with the base theme to create a new theme object
  const getTheme = (mode: string) => merge({}, baseTheme, {
    colors: get(baseTheme.colors.modes, mode, baseTheme.colors),
    isDark: true,
  })

  const theme = darkMode ? getTheme('dark') : baseTheme;

  const pageTransitions = useTransition(location, location => location.pathname, {
    from: { transform: 'translateY(100px)', opacity: 0 },
    enter: { transform: 'translateY(0)', opacity: 1 },
    leave: { transform: 'translateY(100px)', opacity: 0 },
  })

  return (
    <ThemeProvider theme={theme}>
      <NetworkStatusProvider>
        <Wrapper>
          <Normalize />
          <GlobalStyles />
          <ErrorBoundary>
            <Header alignItems="center" flex="none" />
          </ErrorBoundary>
          {
            pageTransitions.map(({ item, props, key }) => (
              <AnimatedWrapper key={key} style={props}>
                <ErrorBoundary>
                  <Suspense fallback={<LoadingBar />}>
                    <Routes routes={routes} location={item} />
                  </Suspense>
                </ErrorBoundary>
              </AnimatedWrapper>
            ))
          }
        </Wrapper>
      </NetworkStatusProvider>
    </ThemeProvider>
  );
}

export default withRouter(App);
