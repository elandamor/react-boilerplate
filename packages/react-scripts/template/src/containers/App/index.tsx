import React, { Suspense } from 'react';
import { RouteComponentProps, withRouter } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';
import { Normalize } from 'styled-normalize';
import get from 'lodash/get';
import merge from 'lodash/merge';
// Components
import { ErrorBoundary, Header, LoadingBar, Routes } from '@app/components';
// Contexts
import NetworkStatusProvider from '@app/contexts/networkStatus.context';
// Routes
import routes from '@app/routes';
// Styles
import { Wrapper } from './styles';

import GlobalStyles from '@app/global-styles';

import baseTheme from '@app/theme';
import { useDarkMode } from '@app/hooks';

// import { makeDebugger } from '@app/utils';
// const debug = makeDebugger('App');

export interface IAppProps extends RouteComponentProps {}

export const AppContext = React.createContext({});

/**
 * @render react
 * @name App container
 * @description The skeleton around the actual pages, and should only
 * contain code that should be seen on all pages. (e.g. navigation bar).
 */

const App = () => {
  const [darkMode, setDarkMode] = useDarkMode();

  // Merge the color mode with the base theme to create a new theme object
  const getTheme = (mode: string) => merge({}, baseTheme, {
    colors: get(baseTheme.colors.modes, mode, baseTheme.colors),
    isDark: true,
  })

  const theme = darkMode ? getTheme('dark') : baseTheme;

  return (
    <ThemeProvider theme={theme}>
      <NetworkStatusProvider>
        <AppContext.Provider value={{ darkMode, setDarkMode }}>
          <Wrapper>
            <Normalize />
            <GlobalStyles />
            <ErrorBoundary>
              <Header alignItems="center" flex="none" />
            </ErrorBoundary>
            <ErrorBoundary>
              <Suspense fallback={<LoadingBar />}>
                <Routes routes={routes} />
              </Suspense>
            </ErrorBoundary>
          </Wrapper>
        </AppContext.Provider>
      </NetworkStatusProvider>
    </ThemeProvider>
  );
}

export default withRouter(App);
