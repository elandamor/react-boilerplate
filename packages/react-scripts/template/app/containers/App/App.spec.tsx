// App.spec.tsx
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import 'jest-styled-components';

import App from './index';

describe('<App />', () => {
  it('should render App without crashing', () => {
    const component = renderer.create(
      <Router>
        <App />        
      </Router>
    ).toJSON();

    expect(component).toMatchSnapshot();
  });
});
