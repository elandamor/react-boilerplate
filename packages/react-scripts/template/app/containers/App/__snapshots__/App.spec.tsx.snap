// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render App without crashing 1`] = `
.c0 {
  background: var(--brand-success,#000000);
  display: none;
  height: 2px;
  left: 0;
  position: fixed;
  right: 0;
  top: 0;
  -webkit-transform: translateX(100%);
  -ms-transform: translateX(100%);
  transform: translateX(100%);
  z-index: 10;
}

.c0.-loading {
  display: block;
  -webkit-animation: jxImEi 1s ease-in-out infinite;
  animation: jxImEi 1s ease-in-out infinite;
  -webkit-animation-delay: .8s;
  animation-delay: .8s;
}

<div
  className="c-app__container v-xsmall "
>
  <div
    className="c-loadingBar -loading c0"
  />
</div>
`;
